### ISSUE TITLE
[DateRangePicker] Input skips first character when manual typing

### ISSUE THREAD

**Issue by vbynda-gresham** [mui/mui-x#5396]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Current behavior 😯

When manually typing into Date range picker input field, first typed character is 'skipped'.

### Expected behavior 🤔

To see all typed characters.

### Steps to reproduce 🕹

Steps:

1. Go https://mui.com/x/react-date-pickers/date-range-picker/#basic-usage
2. Type 12 into the first input field labeled 'Check in'
3. Expected to see 12/, got 2.
4.

### Your environment 🌎

<details>
  <summary><code>npx @mui/envinfo</code></summary>

```
  Don't forget to mention which browser you used.
  Output from `npx @mui/envinfo` goes here.
```
</details>

### Order ID 💳 (optional)
45466

**Comment by cherniavskii** [comment]
It's a regression introduced in 5.0.0-alpha.6:
- 5.0.0-alpha.5 - https://codesandbox.io/s/basicdaterangepicker-demo-mui-x-forked-ycdx8j?file=/package.json
- 5.0.0-alpha.6 - https://codesandbox.io/s/basicdaterangepicker-demo-mui-x-forked-iz5in6?file=/package.json

**Comment by cherniavskii** [comment]
cc @flaviendelangle @alexfauquette

**Comment by alexfauquette** [comment]
Fixed in #5265

should be available in next release
